extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -41,6 +45,24 @@
  */
 @Singleton
 public class HttpAuthenticator {
+    /**
+     * Special handling for the basic authentication mechanism, for user convenience, we add the mechanism when:
+     * - not explicitly disabled or enabled
+     * - is default bean and not programmatically looked up because there are other authentication mechanisms
+     * - no custom auth mechanism is defined because then, we can't tell if user didn't provide custom impl.
+     * - there is a provider that supports it (if not, we inform user via the log)
+     * <p>
+     * Presence of this system property means that we need to test whether:
+     * - there are HTTP Permissions using explicitly this mechanism
+     * - or {@link io.quarkus.vertx.http.runtime.security.annotation.BasicAuthentication}
+     */
+    public static final String TEST_IF_BASIC_AUTH_IMPLICITLY_REQUIRED = "io.quarkus.security.http.test-if-basic-auth-implicitly-required";
```

Comment:
In the linked issue, the problem is that the user has noticed there was no need to enable basic authentication, when no other mechanisms was used - and then was surprised the basic authentication had to be enabled manually even though the HTTP Policy says the basic authentication is required.
So asking users to set this property will make harder to the users, if we know `BASIC_AUTH_ANNOTATION_DETECTED`. Can we avoid it ?
We also have a case of HTTP policy saying Basic is required

Reply:
We are not asking user to do anything. It's just a way to propagate information because creating the HTTP Authenticator as a synthetic bean is not a suitable (at least for now, we need to be extremely careful when is it created as I mentioned that some extensions like Elytron fails if the bean is initialized too early).

User needs to do nothing. I have tests that proves it (and I'll add the one requested by you).

Reply:
> In the linked issue, the problem is that the user has noticed there was no need to enable basic authentication, when no other mechanisms was used - and then was surprised the basic authentication had to be enabled manually even though the HTTP Policy says the basic authentication is required.

this PR solves user issue as shown in the `integration-tests/oidc` module without doing anything